stdlib/LinearAlgebra/src/lapack.jl

```diff
@@ -5030,6 +5030,7 @@ for (syev, syevr, sygvd, elty) in
                         vl::AbstractFloat, vu::AbstractFloat, il::Integer, iu::Integer, abstol::AbstractFloat)
             chkstride1(A)
             n = checksquare(A)
+            chkfinite(A)
```

Comment:
I am not sure, that `chfinite` should be called unmodified. As it checks all elements of `A`, that could erroneously throw exceptions, if there is a non-finite element in the unused part of `A`.
Only one triagle of `A` should be checked, dependant on `uplo`.

Reply:
Good point.

Reply:
Looks like this will require adding another function similar to chkfinite() something like :

`function chkuplofinite(A::AbstractMatrix, uplo::AbstractChar)`

which will then check upper or lower part of matrix for non-finite element according to `uplo`
Should I go ahead and implement this?

Reply:
@dkarrasch @KlausC I've added the functionality as requested.